### DISCUSSION TITLE
Valibot schema validation

### DISCUSSION THREAD

**Discussion by IanVS** [react-hook-form/react-hook-form#10724] [Ideas] [answered]
There is a new, lighterweight alternative to zod that has just been released, called Valibot.  https://www.builder.io/blog/introducing-valibot.  

What would it take to be able to use their schemas with react-hook-form?

**Comment by craigmcc** [comment]
One approach (on a per-form basis) would be to follow the example in the RHF docs:

https://react-hook-form.com/advanced-usage#CustomHookwithResolver

(They used Yup in the example, but you could code it with whatever validation logic you want.)

A more comprehensive approach would be to implement your own resolver so that it can be reused.  The existing ones all live in the "@hookform/resolvers" package on GitHub:

https://github.com/react-hook-form/resolvers

I didn't see any specific how-to instructions for building one, but there's lots of examples there.

**Reply by SalahAdDin** [reply, marked as answer]
Well, it does have a [`valibot` resolver](https://github.com/react-hook-form/resolvers/tree/master/valibot).

**Reply by IanVS** [reply]
Great, thanks for making that, @erikshestopal!

**Reply by erikshestopal** [reply]
@IanVS you're welcome!

**Reply by SalahAdDin** [reply]
@erikshestopal Thank you!

**Reply by naquiroz** [reply]
could you add docs to it? Its not in the resolvers section

**Reply by SalahAdDin** [reply]
> could you add docs to it? It's not in the resolvers section

It is in the repository.

**Reply by naquiroz** [reply]
Ah I yes I found it.
